src/main/java/com/scottlogic/filter/FilterLikedPosts.java

```diff
@@ -0,0 +1,19 @@
+package com.scottlogic.filter;
+
+import com.scottlogic.UserPost;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FilterLikedPosts implements Filter {
+    @Override
+    public List<UserPost> filter(List<UserPost> inputList, String filterCriteria) {
```

Comment:
Hmm, odd that we have a filterCriteria but it is not used.
I get that you have all the info for liked posts from the input list, maybe we could design this interface neater?

For example your filter method on the interface could just take in an inputList.
The criteria can be set via the constructor for the concrete implementation.

That way FilterLikedPosts can just take the input list and filter without any criteria output, and AuthorFilter can take in a string at the constructor then use it in the filter implementation.
The benefit is your criteria won't be confined to a String and you can have as many as you want (from passing them in via the constructor).

What do you think?

Another way would be to implement a FilterCriteria type that would have a type and value, that would allow you to pass multiple criteria objects without string parsing, but would encounter similar issues with changing types.

Reply:
This is something I ran into when implementing multiple filters - I've now got a single argument in 'filter' and a constructor that takes the criteria and sets them into private variables within each filter-class. Will be pushed with the next PR.